src/endpoint/s3/ops/s3_post_object_select.js

```diff
@@ -0,0 +1,206 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const dbg = require('../../../util/debug_module')(__filename);
+const S3Error = require('../s3_errors').S3Error;
+const s3_utils = require('../s3_utils');
+const { Transform } = require('readable-stream');
+const nb_native = require('../../../util/nb_native');
+const crc32 = require('crc-32');
+
+//end message is always the same, so we allocate and init it once.
+const end_message = Buffer.concat([
+    Buffer.from([0, 0, 0, 56]), //PRELUDE begins: total byte length
+    Buffer.from([0, 0, 0, 40]), //headers byte length
+    Buffer.from([0xc1, 0xc6, 0x84, 0xd4]), //crc of above (end of PRELUDE)
+    Buffer.from([11]), //DATA begins, HEADERS begin: byte length of below header name
+    Buffer.from(':event-type'),
+    Buffer.from([7, 0, 3]), //type (7, string) of below header value
+    Buffer.from('End'),
+    Buffer.from([13]), //again for second header
+    Buffer.from(':message-type'),
+    Buffer.from([7, 0, 5]),
+    Buffer.from('event'), //end of HEADERS
+    //no PAYLOAD (DATA ends)
+    Buffer.from([0xfe, 0x2c, 0xee, 0x99]) //MESSAGE CRC (crc of above)
+]);
+
+//same for headers of records message
+const records_message_headers = Buffer.concat([
+    Buffer.from([11]), //byte length of below header name
+    Buffer.from(':event-type'),
+    Buffer.from([7, 0, 7]), //type (7, string) of below header value
+    Buffer.from('Records'),
+    Buffer.from([13]), //again for content-type header
+    Buffer.from(':content-type'),
+    Buffer.from([7, 0, 24]),
+    Buffer.from('application/octet-stream'),
+    Buffer.from([13]), //again for message-type header
+    Buffer.from(':message-type'),
+    Buffer.from([7, 0, 5]),
+    Buffer.from('event')
+]);
+
+class S3SelectStream extends Transform {
+
+    constructor(opts) {
+        super(opts);
+        this.s3select = new (nb_native().S3Select)(opts);
+    }
+
+    async encode_chunk(select_result) {
+        dbg.log2("select res = ", select_result.toString());
+
+        const total_length_bytes = select_result.length + records_message_headers.length + 16;
```

Comment:
Please change the 'magic number' (16) so that it would be easier to maintain the code.